src/shader_compiler.cc

```diff
@@ -54,10 +54,11 @@ ShaderCompiler::ShaderCompiler(const std::string& env,
 ShaderCompiler::~ShaderCompiler() = default;
 
 std::pair<Result, std::vector<uint32_t>> ShaderCompiler::Compile(
+    std::string pipeline_name,
     Pipeline::ShaderInfo* shader_info,
     const ShaderMap& shader_map) const {
   const auto shader = shader_info->GetShader();
-  auto it = shader_map.find(shader->GetName());
+  auto it = shader_map.find(pipeline_name + "-" + shader->GetName());
```

Comment:
For the case where the shader is not in a pipeline, the "-" should not be present I think. It should never happen here as we call this with a pipeline all the time, but we should probably check if pipeline_name is "" to not add the "-".

Reply:
Done